### PR TITLE
Refine embedding Api doc

### PR DESCRIPTION
**Pull request by Aurelius84**
+ Add discription of `input` IDs which should satisfy `0<=id<size[0]`
+ Improved the instructions for using `padding_idx`
embedding

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -373,17 +373,18 @@ def embedding(input,
...
             embeddings and the size of each embedding vector respectively.
         is_sparse(bool): The flag indicating whether to use sparse update.
         is_distributed(bool): Whether to run lookup table from remote parameter server.
-        padding_idx(int|long|None): If :attr:`None`, it makes no effect to lookup.
-            Otherwise the given :attr:`padding_idx` indicates padding the output
-            with zeros whenever lookup encounters it in :attr:`input`. If
-            :math:`padding_idx < 0`, the :attr:`padding_idx` to use in lookup is
-            :math:`size[0] + dim`.
-        param_attr(ParamAttr): Parameters for this layer
+        padding_idx(int|long|None): It will output all-zero paddig data whenever
```
Comment (guoshengCS) [inline python/paddle/fluid/layers/nn.py]: `paddig` 拼写错误

Comment (Aurelius84) [reply python/paddle/fluid/layers/nn.py]: Thanks! Fixed

**Review by guoshengCS — DISMISSED** [review]

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -373,17 +373,18 @@ def embedding(input,
...
-            :math:`padding_idx < 0`, the :attr:`padding_idx` to use in lookup is
-            :math:`size[0] + dim`.
-        param_attr(ParamAttr): Parameters for this layer
+        padding_idx(int|long|None): It will output all-zero padding data whenever
+            lookup encounters :math:`padding\_idx` in Ids. If set :attr:`None`, it makes
+            no effect to output. If :math:`padding\_idx < 0`, the :math:`padding\_idx`
+            will automatically be converted to :math:`size[0] + padding\_idx` to use.
+            Default: None.
+        param_attr(ParamAttr): Parameters for this layer.
         dtype(np.dtype|core.VarDesc.VarType|str): The type of data : float32, float_16, int etc
```
Comment (lanxianghit) [inline python/paddle/fluid/layers/nn.py]: 顺便说明一下这里的dtype吧，这个dtype指定的是输出tensor的数据类型

Comment (Aurelius84) [reply python/paddle/fluid/layers/nn.py]: Fixed.

**Review by lanxianghit — APPROVED** [review]
LGTM

**Review by guoshengCS — APPROVED** [review]

**Pull request merged**